terraform/variables/README.md

```diff
@@ -0,0 +1,205 @@
+# Terraform Variables Example
+
+|Title| Description                                                                          |
+|---|--------------------------------------------------------------------------------------|
+|Pack| Docker                                                                               |
+|Cloud| Local                                                                                |
+|Language| n/a                                                                                  |
+|Docs| [Terraform Cloud](https://developer.hashicorp.com/waypoint/plugins/terraform-cloud)                     |
+
+This example demonstrates sourcing output variables from Terraform cloud for use in Waypoint config.
+
+For example, it's possible to create resources like aws VPCs, ecs clusters, and subnets in Terraform,
+define them as Terraform outputs, and then reference them directly in Waypoint config.
+
+This example is intended to be a minimalist demonstration of referencing Terraform outputs, not a full
+real-world use-case. It creates static Terraform output variables of various types (string, list, and map), and then references
+them in local docker deployment config.
+
+## Terraform Setup
+
+### Org and Workspace Creation
+
+First, we'll need a Terraform Cloud workspace. If you don't have one already, you can create one
+using the Terraform found in `terraform/tfc`. If you already have a Terraform org and workspace,
+skip ahead to the next section.
+
+This module requires three inputs:
+
+- A TFC API token. Follow [this guide](https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/api-tokens)
+to obtain a TFC api token.
+
+- A TFC organization name. This is arbitrary, but needs to be globally unique.
+
+- An email address to use as your TFC org admin contact address.
+
+Run Terraform to create the TFC org and workspace. You'll be prompted for the above inputs
+
+```shell
+$ cd ./terraform/tfc
+$ terraform init
+...
+$ terraform apply
+...
+Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
+```
+
+### Generating Terraform Outputs
+
+In `terraform/main`, we have the minimal possible Terraform to generate ouputs. This is meant to simulate
+real Terraform that might be creating and AWS VPC, and outputting things like VPC IDs and subnet arns.
+
+You'll need the following inputs:
+
+- An org ID and workspace ID. These cannot be referenced as variables, so you'll need to modify 
+`./terraform/main/main.tf`. Use the organization name chosen in the previous step, or set this
+to your own organization ID if desired.
+
+```hcl
+terraform {
+  cloud {
+    organization = "waypoint_tfc_vars_example" // <- you must change this
+
+    workspaces {
+      name = "waypoint_tfc_vars_example"
+    }
+  }
+}
```

Comment:
I think we could make both of these input variables. Although if it's a set once and forget, maybe not worth it for the example!

Reply:
I tried! Input variables aren't allowed in this block 😭

Reply:
Oooh shoot! That's true, input vars don't work everywhere in the hcl file.